front/lib/api/assistant/configuration.ts

```diff
@@ -75,6 +84,14 @@ export async function getAgentConfiguration(
         model: AgentDustAppRunConfiguration,
         as: "dustAppRunConfiguration",
       },
+      {
+        model: AgentUserRelation,
```

Comment:
Trusting you on this query. We're on a performance critical path so this has to be super efficient.

Reply:
Checking the indexes I thiiiink it's covered but would be nice to EXPLAIN it to get a cost :+1:

Reply:
I had made the theoretical check beforehand. I would be surprised and disappointed by Sequelize (or postgres) if it tdidnt' work but ok to check in practice.
```
 Nested Loop Left Join  (cost=0.71..41.88 rows=1 width=7508) (actual time=0.060..0.063 rows=1 loops=1)
   ->  Nested Loop Left Join  (cost=0.56..33.70 rows=1 width=6988) (actual time=0.047..0.049 rows=1 loops=1)
         ->  Nested Loop Left Join  (cost=0.42..25.23 rows=1 width=5420) (actual time=0.041..0.043 rows=1 loops=1)
               ->  Nested Loop Left Join  (cost=0.28..16.56 rows=1 width=2780) (actual time=0.034..0.036 rows=1 loops=1)
                     ->  Limit  (cost=0.14..8.16 rows=1 width=1688) (actual time=0.022..0.023 rows=1 loops=1)
                           ->  Index Scan Backward using agent_configurations_s_id_version on agent_configurations agent_configuration  (cost=0.14..8.16 rows=1 width=1688) (actual time=0.020..0.020 rows=1 loops=1)
                                 Index Cond: (("sId")::text = 'f3433c6e46'::text)
                                 Filter: ("workspaceId" = 1)
                     ->  Index Scan using agent_generation_configurations_pkey on agent_generation_configurations "generationConfiguration"  (cost=0.14..8.16 rows=1 width=1092) (actual time=0.008..0.008 rows=1 loops=1)
                           Index Cond: (id = agent_configuration."generationConfigurationId")
               ->  Index Scan using agent_retrieval_configurations_pkey on agent_retrieval_configurations "retrievalConfiguration"  (cost=0.14..8.15 rows=1 width=2640) (actual time=0.005..0.005 rows=0 loops=1)
                     Index Cond: (id = agent_configuration."retrievalConfigurationId")
         ->  Index Scan using agent_dust_app_run_configurations_pkey on agent_dust_app_run_configurations "dustAppRunConfiguration"  (cost=0.14..8.16 rows=1 width=1568) (actual time=0.005..0.005 rows=0 loops=1)
               Index Cond: (id = agent_configuration."dustAppRunConfigurationId")
   ->  Index Scan using agent_user_relation_config_workspace_user_idx on agent_user_relations  (cost=0.14..8.16 rows=1 width=524) (actual time=0.011..0.011 rows=0 loops=1)
         Index Cond: (("agentConfigurationId" = agent_configuration.id) AND ("userId" = 1))
 Planning Time: 1.593 ms
 Execution Time: 0.209 ms
```
Planning seems good (indexed grab of agent user relation)

Reply:
The query for the record
```
EXPLAIN ANALYZE SELECT "agent_configuration".*, "generationConfiguration"."id" AS "generationConfiguration.id", "generationConfiguration"."createdAt" AS "generationConfiguration.createdAt", "generationConfiguration"."updatedAt" AS "generationConfiguration.updatedAt", "generationConfiguration"."prompt" AS "generationConfiguration.prompt", "generationConfiguration"."providerId" AS "generationConfiguration.providerId", "generationConfiguration"."modelId" AS "generationConfiguration.modelId", "generationConfiguration"."temperature" AS "generationConfiguration.temperature", "retrievalConfiguration"."id" AS "retrievalConfiguration.id", "retrievalConfiguration"."createdAt" AS "retrievalConfiguration.createdAt", "retrievalConfiguration"."updatedAt" AS "retrievalConfiguration.updatedAt", "retrievalConfiguration"."sId" AS "retrievalConfiguration.sId", "retrievalConfiguration"."query" AS "retrievalConfiguration.query", "retrievalConfiguration"."queryTemplate" AS "retrievalConfiguration.queryTemplate", "retrievalConfiguration"."relativeTimeFrame" AS "retrievalConfiguration.relativeTimeFrame", "retrievalConfiguration"."relativeTimeFrameDuration" AS "retrievalConfiguration.relativeTimeFrameDuration", "retrievalConfiguration"."relativeTimeFrameUnit" AS "retrievalConfiguration.relativeTimeFrameUnit", "retrievalConfiguration"."topK" AS "retrievalConfiguration.topK", "retrievalConfiguration"."topKMode" AS "retrievalConfiguration.topKMode", "dustAppRunConfiguration"."id" AS "dustAppRunConfiguration.id", "dustAppRunConfiguration"."createdAt" AS "dustAppRunConfiguration.createdAt", "dustAppRunConfiguration"."updatedAt" AS "dustAppRunConfiguration.updatedAt", "dustAppRunConfiguration"."sId" AS "dustAppRunConfiguration.sId", "dustAppRunConfiguration"."appWorkspaceId" AS "dustAppRunConfiguration.appWorkspaceId", "dustAppRunConfiguration"."appId" AS "dustAppRunConfiguration.appId", "agent_user_relations"."id" AS "agent_user_relations.id", "agent_user_relations"."relation" AS "agent_user_relations.relation" FROM (SELECT "agent_configuration"."id", "agent_configuration"."createdAt", "agent_configuration"."updatedAt", "agent_configuration"."sId", "agent_configuration"."version", "agent_configuration"."status", "agent_configuration"."scope", "agent_configuration"."name", "agent_configuration"."description", "agent_configuration"."pictureUrl", "agent_configuration"."workspaceId", "agent_configuration"."generationConfigurationId", "agent_configuration"."retrievalConfigurationId", "agent_configuration"."dustAppRunConfigurationId", "agent_configuration"."authorId" FROM "agent_configurations" AS "agent_configuration" WHERE "agent_configuration"."sId" = 'f3433c6e46' AND "agent_configuration"."workspaceId" = 1 ORDER BY "agent_configuration"."version" DESC LIMIT 1) AS "agent_configuration" LEFT OUTER JOIN "agent_generation_configurations" AS "generationConfiguration" ON "agent_configuration"."generationConfigurationId" = "generationConfiguration"."id" LEFT OUTER JOIN "agent_retrieval_configurations" AS "retrievalConfiguration" ON "agent_configuration"."retrievalConfigurationId" = "retrievalConfiguration"."id" LEFT OUTER JOIN "agent_dust_app_run_configurations" AS "dustAppRunConfiguration" ON "agent_configuration"."dustAppRunConfigurationId" = "dustAppRunConfiguration"."id" LEFT OUTER JOIN "agent_user_relations" AS "agent_user_relations" ON "agent_configuration"."id" = "agent_user_relations"."agentConfigurationId" AND "agent_user_relations"."userId" = 1 ORDER BY "agent_configuration"."version" DESC;
```

Reply:
cost looks good indeed :+1: